service/app/settlement.go

```diff
@@ -51,6 +60,16 @@ func (s *SettlementCollectible) BeforeCreate(tx *gorm.DB) (err error) {
 	return nil
 }
 
+func (s *SettlementCollectible) SetSettled() (err error) {
+	if s.Settled {
+		return fmt.Errorf("settlement collectible already settled")
+	}
+
+	s.Settled = true
```

Comment:
This caused some confusion, since both `SettlementCollectible` and `Settlement` structs have a `Settled` field with differing types (`bool` vs. `uint`). I'd consider renaming the `bool` field to `IsSettled` or using `SettledCount` (and `TotalCount`)  for the `uint` type. Just my opinion though :)

Reply:
Will fix, thank you!